cmake/ZeekSpicyAnalyzerSupport.cmake

```
@@ -6,32 +6,65 @@
 
 include(GNUInstallDirs)
 
-# Add target to build an analyzer. Arguments are the name of the analyzer and a
-# variable number of source files for `spicyz`.
-function(spicy_add_analyzer name)
-    set(sources "${ARGN}")
-    string(TOLOWER "${name}" name_lower)
-    set(output "${SPICY_MODULE_OUTPUT_DIR_BUILD}/${name_lower}.hlto")
-    set(output_install "${SPICY_MODULE_OUTPUT_DIR_INSTALL}/${name_lower}.hlto")
-    set(deps "spicyz")
+# Add target to build an analyzer.
+#
+# Usage:
+#
+#     spicy_add_analyzer(
+#         NAME name
+#         [SOURCES <source files for spicyz>...]
+#         [SCRIPTS <additional script files to install>...]
+#     )
+function(spicy_add_analyzer)
+    set(options)
+    set(oneValueArgs NAME)
+    set(multiValueArgs SOURCES SCRIPTS)
+
+    cmake_parse_arguments(PARSE_ARGV 0 SPICY_ANALYZER "${options}" "${oneValueArgs}" "${multiValueArgs}")
+
+    # We also support the legacy behavior where the first arg is
+    # the analyzer NAME and all remaining arguments are SOURCES.
+    if (SPICY_ANALYZER_UNPARSED_ARGUMENTS)
```

Formatting nit:

```suggestion
    if ( SPICY_ANALYZER_UNPARSED_ARGUMENTS )
```

Similar below.